ДЗ Спринт №20

CasTeTRU opened the pull request.

CasTeTRU committed:
ДЗ Спринт №20

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/search/data/TrackRepositoryImpl.kt:
```diff
@@ -5,20 +5,23 @@ import com.example.playlistmaker.search.domain.Track
-            val results = response.body()?.results?.map { it.toDomain() } ?: emptyList()
+    override fun searchTracks(query: String): Flow<List<TrackDomainModel>> = flow {
+        try {
+            android.util.Log.d("TrackRepositoryImpl", "Making API request for query: $query")
```
Дебажные логи лучше удалять из кода, чтобы не засорять код

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/search/data/TrackRepositoryImpl.kt:
```diff
@@ -5,20 +5,23 @@ import com.example.playlistmaker.search.domain.Track
-            emptyList()
+            emit(results)
+        } catch (e: Exception) {
+            android.util.Log.e("TrackRepositoryImpl", "API request failed", e)
```
Путь до класса лучше заменить на корректный импорт

ArturNurtdinov reviewed (commented):
1. Нужно поправить вёрстку экрана настроек - размер текста, отступы между текстом:

<img width="567" height="723" alt="image" src="https://github.com/user-attachments/assets/6c3c5e43-f4e9-4359-8d28-c14c63bc9e72" />

Макет:

<img width="671" height="629" alt="image" src="https://github.com/user-attachments/assets/0d773145-7e04-47eb-ac69-7190f5d06af3" />

2. Нужно поправить цвет невыделенных элементов в нижней панели навигации:

<img width="568" height="143" alt="image" src="https://github.com/user-attachments/assets/8a66e4fd-60a0-41ae-bad7-8b2a9ef43ff0" />

Макет:

<img width="673" height="198" alt="image" src="https://github.com/user-attachments/assets/b86ee63c-30dd-40fc-9396-4d2e973386e9" />

3. Нужно исправить выделение активного таба в TabLayout на экране медиатеки:

<img width="566" height="637" alt="image" src="https://github.com/user-attachments/assets/f602313e-1b45-4129-83fa-d4f597caed93" />

Макет:

<img width="571" height="680" alt="image" src="https://github.com/user-attachments/assets/93e4747d-6d7b-4439-b583-2335140da641" />

4. Нужно убрать отображение истории поиска при пустой истории поиска:

<img width="355" height="790" alt="image" src="https://github.com/user-attachments/assets/9279ae1c-1c37-40dd-ac6e-d2039a81cbc4" />

5. Нужно исправить цвет текста в поле ввода в тёмной теме:

<img width="802" height="298" alt="image" src="https://github.com/user-attachments/assets/c0c91561-f06b-46bb-bb6a-789a24958cd8" />

Макет:

<img width="892" height="307" alt="image" src="https://github.com/user-attachments/assets/bd4fe6e0-ec32-4eef-bbb9-22e0e24538cc" />

6. Нужно исправить отступ списка треков снизу от нижней панели навигации:

<img width="750" height="697" alt="image" src="https://github.com/user-attachments/assets/03ed83a7-2102-4efc-9800-1f68a4a67033" />

7. При выполнении поискового запроса должен отображаться ProgressBar:

<img width="797" height="673" alt="image" src="https://github.com/user-attachments/assets/95fdc754-d562-420b-a13b-80096bcd5131" />

CasTeTRU committed:
Отработаны замечания по спринту 20:
1. Нужно поправить цвет невыделенных элементов в нижней панели навигации

2. Нужно исправить отступ списка треков снизу от нижней панели навигации

3.При выполнении поискового запроса должен отображаться ProgressBar

4.Нужно убрать отображение истории поиска при пустой истории поиска
5. Нужно исправить цвет текста в поле ввода в тёмной теме

CasTeTRU committed:
Отработаны замечания:
1) Нужно поправить вёрстку экрана настроек - размер текста, отступы между текстом:
2) Нужно исправить выделение активного таба в TabLayout на экране медиатеки (получилось только для темной темы)
3) Дебажные логи лучше удалять из кода, чтобы не засорять код
4) Путь до класса лучше заменить на корректный импорт

ArturNurtdinov reviewed (commented):
1. Экран настроек не соответствует макетам. Нужно поправить вёрстку экрана настроек - размер текста, отступы между текстом:

<img width="568" height="654" alt="image" src="https://github.com/user-attachments/assets/91f240d6-76e3-4511-a9dd-79cbe0a5bb20" />

Макет:

<img width="759" height="723" alt="image" src="https://github.com/user-attachments/assets/691bc7be-2a47-47bb-b989-a9a3c1e91576" />

2. TabLayout не соответствует макетам в светлой теме. Нужно исправить выделение активного таба в TabLayout на экране медиатеки:

<img width="853" height="846" alt="image" src="https://github.com/user-attachments/assets/a7ca722d-7cce-493a-a286-44d39ea1613e" />

Макет:

<img width="775" height="797" alt="image" src="https://github.com/user-attachments/assets/50e9051c-f4f3-4d4d-bfe6-43602731bea6" />

3. При пустом поле вводе отображается плейсхолдер "Ничего не найдено", даже если запрос не выполнялся:

<img width="568" height="694" alt="image" src="https://github.com/user-attachments/assets/7f598b98-3be6-4dce-8fbb-efd0b3a2b3cb" />

4. Кнопка "Очистить" в поле ввода на экране поиска не отображается:

<img width="801" height="296" alt="image" src="https://github.com/user-attachments/assets/4eebfa9d-1e84-471c-9806-08f03ff95fcd" />

CasTeTRU committed:
Отработаны замечания:
1) Экран настроек не соответствует макетам. Нужно поправить вёрстку экрана настроек - размер текста, отступы между текстом
2) TabLayout не соответствует макетам в светлой теме. Нужно исправить выделение активного таба в TabLayout на экране медиатеки
3) При пустом поле вводе отображается плейсхолдер "Ничего не найдено", даже если запрос не выполнялся
4) Кнопка "Очистить" в поле ввода на экране поиска не отображается

CasTeTRU committed:
ДЗ Спринт 20
Применён шрифт YS Display Medium к заголовкам

CasTeTRU commented:
Указанные замечания поправил, единственное что изменения по макету на экране настроек я делал в отдельной ветке (для спринта №21) и не могу их применить на текущий ПР (может быть проблема в IDE, может ещё в чём-то, но одни и те же файлы обработаны по-разному). Прикладываю скриншоты с учтенными замечаниями для экрана настроек. Этот экран уже будет в проекте  при сдаче ДЗ№21
<img width="1440" height="3120" alt="Screenshot_20251103_210350" src="https://github.com/user-attachments/assets/2041490e-d32f-4007-a6c1-607cea8e90dd" />
<img width="1440" height="3120" alt="Screenshot_20251103_210336" src="https://github.com/user-attachments/assets/f38695e0-6864-42aa-b36f-0c9a64aeef04" />
